frontend/public/components/operator-lifecycle-manager/descriptors/README.md

```diff
@@ -0,0 +1,95 @@
+# Operator Descriptors
+
+The Operator Lifecycle Manager allows adding descriptors to Custom Resource Definitions, which are used to describe different fields on the object and actions that can be peformed.
+
+There are three types of descriptors:
+
+Type   | Directory | Description
+-------|-----------|------------
+Spec   | `spec/`   | References fields in the `spec` block of an object
+Status | `status/` | References fields in the `status` block of an object
+Action | `action/` | References actions that can be performed on an object
+
+The schema for a descriptor is the same, regardless of type:
+
+```typescript
+type Descriptor = {
+  path: string; // Dot-delimited path of the field on the object that this descriptor 
+  displayName: string;
+  description: string;
+  'x-descriptors': SpecCapability[] | StatusCapability[]; // Used to determine which "capabilities" this descriptor has, and which React component to use
+  value?: any; // Optional value 
+}
+```
+
+### Example
+
+From the `ClusterServiceVersion` for [etcd-operator](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/deploy/chart/catalog_resources/ocs/etcdoperator.v0.9.2.clusterserviceversion.yaml):
+
+```yaml
+- name: etcdclusters.etcd.database.coreos.com
+  version: v1beta2
+  kind: EtcdCluster
+  displayName: etcd Cluster
+  description: Represents a cluster of etcd nodes.
+  resources:
+    - kind: Service
+      version: v1
+    - kind: Pod
+      version: v1
+  specDescriptors:
+    - description: The desired number of member Pods for the etcd cluster.
+      displayName: Size
+      path: size
+      x-descriptors:
+        - 'urn:alm:descriptor:com.tectonic.ui:podCount'
```

Comment:
Do we want to take this opportunity to deprecate `alm` and replace it with `olm`? We'd still need to support `alm` for back-compat, but we could indicate to users not to use it